// react-native-easy-calendar/lib/commonjs/Components/Arrow.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _Icons = _interopRequireDefault(require("../Icons"));

var _Contexts = require("../Contexts");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Arrow = ({
  direction,
  isDisabled,
  onPress
}) => {
  const theme = _react.default.useContext(_Contexts.ThemeContext);

  return /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    hitSlop: {
      top: 20,
      bottom: 20,
      left: 20,
      right: 20
    },
    accessibilityLabel: "".concat(direction, " arrow"),
    accessibilityHint: 'Press to move to previous month or year',
    accessibilityState: {
      disabled: isDisabled
    },
    disabled: isDisabled,
    onPress: onPress,
    style: [theme.normalArrowContainer, isDisabled && theme.disabledArrowContainer]
  }, /*#__PURE__*/_react.default.createElement(_reactNative.Image, {
    testID: 'arrow-image',
    accessibilityIgnoresInvertColors: true,
    source: _Icons.default.arrow.left['16px'],
    style: [theme.normalArrowImage, isDisabled && theme.disabledArrowImage, direction === 'right' && styles.right]
  }));
};

var _default = /*#__PURE__*/_react.default.memo(Arrow);

exports.default = _default;

const styles = _reactNative.StyleSheet.create({
  right: {
    transform: [{
      rotate: '180deg'
    }]
  }
});
