// react-native-easy-calendar/lib/module/Components/Arrow.js
import React from 'react';
import { Image, TouchableOpacity, StyleSheet } from 'react-native';
import Icons from '../Icons';
import { ThemeContext } from '../Contexts';

const Arrow = ({
  direction,
  isDisabled,
  onPress
}) => {
  const theme = React.useContext(ThemeContext);
  return /*#__PURE__*/React.createElement(TouchableOpacity, {
    hitSlop: {
      top: 20,
      bottom: 20,
      left: 20,
      right: 20
    },
    accessibilityLabel: "".concat(direction, " arrow"),
    accessibilityHint: 'Press to move to previous month or year',
    accessibilityState: {
      disabled: isDisabled
    },
    disabled: isDisabled,
    onPress: onPress,
    style: [theme.normalArrowContainer, isDisabled && theme.disabledArrowContainer]
  }, /*#__PURE__*/React.createElement(Image, {
    testID: 'arrow-image',
    accessibilityIgnoresInvertColors: true,
    source: Icons.arrow.left['16px'],
    style: [theme.normalArrowImage, isDisabled && theme.disabledArrowImage, direction === 'right' && styles.right]
  }));
};

export default /*#__PURE__*/React.memo(Arrow);
const styles = StyleSheet.create({
  right: {
    transform: [{
      rotate: '180deg'
    }]
  }
});
