// react-native-easy-calendar/lib/commonjs/Calendars/DateSelectionCalendar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _dayjs = _interopRequireDefault(require("dayjs"));

var _utc = _interopRequireDefault(require("dayjs/plugin/utc"));

var _Providers = _interopRequireDefault(require("./Providers"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

_dayjs.default.extend(_utc.default);

// A thin wrapper to limit the props that can be passed to the BaseCalendar component
const DateSelectionCalendar = ({
  onSelectDate,
  disabledDates,
  selectedDate,
  initVisibleDate,
  allowYearView = true,
  ...others
}) => {
  if (!selectedDate) {
    throw new Error('The `selectedDate` prop is required. Use an empty array if no dates should be selected.');
  }

  if (typeof selectedDate !== 'string') {
    throw new Error('The `selectedDate` prop should be a date string in YYYY-MM-DD format.');
  }

  if (!onSelectDate) {
    throw new Error('The `onSelectDate` prop is required.');
  }

  if (typeof onSelectDate !== 'function') {
    throw new Error('The `onSelectDate` prop should be function that receives a date string as paramater.');
  }

  const dateProperties = (0, _react.useMemo)(() => {
    let disabledDateProperties = {};
    let selectedDateProperties = {};
    disabledDateProperties = disabledDates === null || disabledDates === void 0 ? void 0 : disabledDates.reduce((disabled, date) => {
      disabled[date] = {
        isDisabled: true
      };
      return disabled;
    }, {});

    if ((0, _dayjs.default)(selectedDate).isValid()) {
      selectedDateProperties = {
        [(0, _dayjs.default)(selectedDate).local().format('YYYY-MM-DD')]: {
          isSelected: true
        }
      };
    } // Not possible for a date to be both disabled and selected, so overwriting is OK


    return { ...disabledDateProperties,
      ...selectedDateProperties
    };
  }, [selectedDate, disabledDates]);
  return /*#__PURE__*/_react.default.createElement(_Providers.default, _extends({
    allowYearView: allowYearView,
    onPressDay: onSelectDate,
    initVisibleDate: initVisibleDate || ((0, _dayjs.default)(selectedDate).isValid() ? selectedDate : undefined),
    dateProperties: dateProperties
  }, others));
};

var _default = DateSelectionCalendar;
exports.default = _default;
