// react-native-easy-calendar/lib/commonjs/Calendars/MultiDateSelectionCalendar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _dayjs = _interopRequireDefault(require("dayjs"));

var _utc = _interopRequireDefault(require("dayjs/plugin/utc"));

var _Providers = _interopRequireDefault(require("./Providers"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

_dayjs.default.extend(_utc.default);

// A thin wrapper to limit the props that can be passed to the BaseCalendar component
const MultiDateSelectionCalendar = ({
  onSelectDates,
  disabledDates,
  selectedDates,
  allowYearView = true,
  ...others
}) => {
  if (!selectedDates) {
    throw new Error('The `selectedDates` prop is required. Use an empty array if no dates should be selected.');
  }

  if (typeof selectedDates !== 'object') {
    throw new Error('The `selectedDates` prop should be an array of date strings in YYYY-MM-DD format.');
  }

  if (!onSelectDates) {
    throw new Error('The `onSelectDates` prop is required.');
  }

  if (typeof onSelectDates !== 'function') {
    throw new Error('The `onSelectDates` prop should be function that receives an array of date strings as paramater.');
  }

  const selDatesRef = _react.default.useRef(selectedDates);

  const dateProperties = (0, _react.useMemo)(() => {
    const disabledDateProperties = disabledDates === null || disabledDates === void 0 ? void 0 : disabledDates.reduce((disabled, date) => {
      disabled[date] = {
        isDisabled: true
      };
      return disabled;
    }, {});
    const selectedDateProperties = selectedDates.reduce((selected, date) => {
      selected[date] = {
        isSelected: true
      };
      return selected;
    }, {}); // Not possible for a date to be both disabled and selected, so overwriting is OK

    return { ...disabledDateProperties,
      ...selectedDateProperties
    };
  }, [selectedDates, disabledDates]);

  const remove = dateToRemove => {
    const newSelectedDates = selDatesRef.current.filter(date => date !== dateToRemove);
    selDatesRef.current = newSelectedDates;
    return newSelectedDates;
  };

  const append = dateToAppend => {
    const newSelectedDates = [...selDatesRef.current, dateToAppend].sort();
    selDatesRef.current = newSelectedDates;
    return newSelectedDates;
  };

  const onPressDay = _react.default.useCallback(date => {
    if (selDatesRef.current.includes(date)) {
      onSelectDates(remove(date));
    } else {
      onSelectDates(append(date));
    }
  }, [onSelectDates]);

  return /*#__PURE__*/_react.default.createElement(_Providers.default, _extends({
    onPressDay: onPressDay,
    allowYearView: allowYearView,
    dateProperties: dateProperties
  }, others));
};

var _default = MultiDateSelectionCalendar;
exports.default = _default;
